collect_app/src/main/java/org/odk/collect/android/application/initialization/ApplicationInitializer.java

```diff
@@ -69,6 +69,8 @@ public void initialize() {
     private void performUpgradeIfNeeded() {
         if (launchState.isUpgradedFirstLaunch()) {
             appUpgrader.upgrade();
+        } else {
+            appUpgrader.install();
```

Comment:
Doesn't it mean that if you have v2021.3 and install v2021.4 (assuming we are going to release it with this fix) it will still go and perform `appUpgrader.upgrade()` first and run those migrations?

Reply:
I'm pretty sure I'm right...

Reply:
Errr maybe? Just going to try and write this out:

* Fresh install of 2021.3.2 (or any version with projects) and upgrade to 2022.1: upgrade() runs without ever running install() so they get an extra project 👎
* Fresh install of 2021.3.3 (next hot fix) and upgrade to 2022.1: upgrade() runs but after install() on fresh install 👍
* Upgrade to 2021.3.3 from 2021.3.2 with previous upgrade from 2021.3.1: upgrade() runs but after ExistingProjectMIgrator has already run on previous upgrades 👍

So yeah I think you're right that this hasn't quite fixed the problem. If we'd included `install()` in our original release with `ExistingProjectMigrator` everything would work but as you point out, we need to account for the existing bad state. 

I think to correct the problem we need to run `install()` before `upgrade`() when upgrading from a version before 2021.3.3 (our next hot fix) and after 1.30.1 (all versions after this have projects and `ExistingProjectMigrator`). Does that sound right? We'll need to work out version codes for these as well.

Reply:
Can't you just base your fix on `LAST_LAUNCHED` value that we keep in meta settings? If that value represents a version that already had projects the migration should be skipped.

Reply:
Yeah sorry I should have worded that differently, calling `install()` is a bit over the top. How about: "stop `ExistingProjectMigrator` from running when upgrading from a version after 1.30.1 (using version code to determine the last version)". Or actually, even simpler: "stop `ExistingProjectMigrator` running when upgrading from a version without `LAST_LAUNCHED`" as `LAST_LAUNCHED` was introduced as part of 2021.2.

Reply:
If `LAST_LAUNCHED`  was released in 2021.2 it won't be a perfect fix because still upgrading from 2021.0 and 2021.1 will be broken but that doesn't seem to be a big deal. Alternatively you can think of another solution like for example:
`If projects repository is not empty that means we do upgrade from a version with projects and can skip migrations.`

Reply:
There's no 2021.0 or .1. 1.30 was the last version before projects and 2021.2 first version with them. We **want** upgrading from 1.30 (and before) to run `ExistingProjectMigrator` and there is already the upgrade `key` concept that will take care of making sure that the upgrade doesn't run again after that, so I think we're fine on older versions.

Reply:
Right my mistake then ok it will be fine.